### PR TITLE
KPCF v1.1.0 publishing

### PR DESCRIPTION
**Pull request by Dubjunk**

### CODE REVIEW

File: KP-Cratefiller/KPCF/fnc/fn_getItems.sqf [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+} forEach [(configFile >> "CfgMagazines"), (configFile >> "CfgWeapons"), (configFile >> "CfgVehicles"), (configFile >> "CfgGlasses")];
+
+// Convert to classname
+{
+    _classNames pushBack (configName _x)
+} forEach _configClasses;
+
+// Black & whitelisting
+_classNames = _classNames - KPCF_blacklistedItems;
+_classNames append KPCF_whitelistedItems;
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_getItems.sqf]: Shouldn't it be only one or another? 🤔 

If you get `_classnames` by crawling, shouldn't it already have all whitelistedItems registered?

Maybe something like (pseudo code)
If elements in whitelistedItems, then _classnames = whitelistedItems
else crawl the configs to get all available items and remove everything which is listed in blacklistedItems

File: KP-Cratefiller/KPCF/fnc/fn_getItems.sqf [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            switch (_type select 1) do {
+                case "AccessoryMuzzle" : {};
+                case "AccessoryPointer" : {};
+                case "AccessorySights" : {};
+                case "AccessoryBipod" : {};
+                default {KPCF_items pushBackUnique _x};
+            };
+        };
+    };
+} forEach _classNames;
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_getItems.sqf]: Couldn't directly say if it's really more performance friendly or if `pushBackUnique` is really noticeable performance heavier then `pushBack`, but you could think about:

As it seems you expecting to have some doubled classnames in the array (as you use `pushBackUnique`) you could sort out every doubled, etc. entries before the forEach or as forEach argument with
`forEach (_classnames arrayIntersect _classnames);` and use just `pushBack` in the switch.

Ref (example 2): https://community.bistudio.com/wiki/arrayIntersect

File: KP-Cratefiller/KPCF/fnc/fn_import.sqf [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+private _importName = _ctrlImport lbText _index;
+
+// Check for empty selection
+if (_index == -1) exitWith {
+    hint localize "STR_KPCF_HINTSELECTION";
+    [{hintSilent "";}, [], 3] call CBA_fnc_waitAndExecute;
+};
+
+// Read the presets from profileNamespace
+private _import = profileNamespace getVariable "KPCF_preset";
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_import.sqf]:
```suggestion
private _import = profileNamespace getVariable ["KPCF_preset", []];
```

File: KP-Cratefiller/KPCF/fnc/fn_manageAceActions.sqf [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    Adds actions to the cratefiller objects.
+
+    Parameter(s):
+    0 : OBJECT - defines the object to apply the action
+
+    Returns:
+    NONE
+*/
+
+params ["_cfBase"];
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_manageAceActions.sqf]: I would suggest to be as precise as you can be concerning parameters etc.
This way it's much easier to debug later on, if it happens that something or someone calls your function with not intended or catched datatypes. (suggestion applies to all params usages, not just this particular one)
```suggestion
params [
    ["_cfBase", objNull, [objNull]]
];
```

File: KP-Cratefiller/KPCF/fnc/fn_manageActions.sqf [hunk 5]
```diff
@@ -8,24 +8,16 @@
...
-} forEach KPCF_cratefillerBase;
-
-// Add the action
-{
-    if (!(_x in KPCF_usedBase)) then {
-        _x addAction ["<t color='#FF8000'>" + localize "STR_KPCF_ACTIONOPEN" + "</t>", {[_this] call KPCF_fnc_openDialog;}, nil, 1, false, true, "", "true", KPCF_interactRadius];
-        KPCF_usedBase pushBack _x;
-    };
-} forEach _foundBase;
+params ["_cfBase"];
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_manageActions.sqf]:
```suggestion
params [
    ["_cfBase", objNull, [objNull]]
];
```

File: KP-Cratefiller/KPCF/fnc/fn_sortList.sqf [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    Parameter(s):
+    0 : ARRAY - Array of classnames
+
+    Returns:
+    0 : ARRAY - Array of sorted subarrays with displayName and classname
+*/
+
+params [
+    "_list"
```
Comment (Wyqer) [inline KP-Cratefiller/KPCF/fnc/fn_sortList.sqf]:
```suggestion
    ["_list", [], [[]]]
```

File: KP-Cratefiller/KPCF/fnc/fn_getItems.sqf (hunk 1 above)
Comment (Dubjunk) [reply KP-Cratefiller/KPCF/fnc/fn_getItems.sqf]: Whitelisting is implemented as some kind of security feature, thought about items that won't be recognized like they should.

File: KP-Cratefiller/KPCF/fnc/fn_getItems.sqf (hunk 2 above)
Comment (Dubjunk) [reply KP-Cratefiller/KPCF/fnc/fn_getItems.sqf]: nice idea, implemented it like that !

File: KP-Cratefiller/KPCF/fnc/fn_getItems.sqf (hunk 1 above)
Comment (Wyqer) [reply KP-Cratefiller/KPCF/fnc/fn_getItems.sqf]: Ah, I see.
As far as I know you've described it like that also in the config file. 👍

**Pull request merged**